"Transportr keeps stopping" upon refreshing connection details

**To Reproduce**

* Open Transportr
* Enter some destination into "Search here", e.g. "Hauptbahnhof, Frankfurt (Main)" in the RMV network
* Click on red circle at bottom right
* Enter some departure into "From", e.g. "Hauptbahnhof, Mainz"
* Click on some connection to get to the details page
* Go back to the list
* Go to the connection details page
* Refresh the connection

**Expected behavior**

* The connection gets refreshed

**Actual behavior**

Case 1:

* Transportr force closes with stacktrace number 1

Case 2:

* Upon refreshing, the user get bumped back to an empty connections list with empty from field (#706) and with the loading animation going on forever (stacktrace number 2)

**Versions**

 - **Transportr Version:** 2.1.0
 - **Device:** LG-D802
 - **Android Version:** 10
 - **LineageOS Version:** 17.1

**Logcat output**

Here are the outputs of _adb logcat_.

**Stacktrace number 1**

```bash
$ adb logcat | grep -F "`adb shell ps | grep de.grobox.  | tr -s [:space:] ' ' | cut -d' ' -f2`"
09-09 18:15:59.435  1435  1435 D Zygote  : Forked child process 5269
09-09 18:15:59.438  1461  1721 I ActivityManager: Start proc 5269:de.grobox.liberario/u0a133 for pre-top-activity {de.grobox.liberario/de.grobox.transportr.map.MapActivity}
09-09 18:15:59.541  5269  5269 V AndroidLocationEngine: Initializing.
09-09 18:15:59.542  5269  5269 V MapboxTelemetry: Initializing telemetry.
09-09 18:15:59.552  5269  5269 D NetworkSecurityConfig: Using Network Security Config from resource network_security_config debugBuild: false
09-09 18:15:59.575  5269  5269 V MapboxTelemetry: Disabling telemetry.
09-09 18:15:59.576  5269  5269 D MapboxTelemetry: Shutting down telemetry service.
09-09 18:15:59.580  5269  5269 I robox.liberari: Thread[1,tid=5269,Native,Thread*=0xb3100e00,peer=0x723211f0,"main"] recursive attempt to load library "/data/app/de.grobox.liberario-KBS5ZhV_anmfK-co2N5C-Q==/lib/arm/libmapbox-gl.so"
09-09 18:15:59.585  5269  5269 I chatty  : uid=10133(de.grobox.liberario) identical 1 line
09-09 18:15:59.585  5269  5269 I robox.liberari: Thread[1,tid=5269,Native,Thread*=0xb3100e00,peer=0x723211f0,"main"] recursive attempt to load library "/data/app/de.grobox.liberario-KBS5ZhV_anmfK-co2N5C-Q==/lib/arm/libmapbox-gl.so"
09-09 18:15:59.586  5269  5269 V MapboxTelemetry: Telemetry was already initialized on that state (enabled: false).
09-09 18:15:59.653  5269  5269 E ConstraintLayout: layout_constraintWidth_default="wrap" is deprecated.
09-09 18:15:59.653  5269  5269 E ConstraintLayout: Use layout_width="WRAP_CONTENT" and layout_constrainedWidth="true" instead.
09-09 18:15:59.783  5269  5296 I Adreno-EGL: <qeglDrvAPI_eglInitialize:379>: EGL 1.4 QUALCOMM build: Nondeterministic_AU_msm8974_LA.BF.1.1.3_RB1__release_AU (I741a3d36ca)
09-09 18:15:59.783  5269  5296 I Adreno-EGL: OpenGL ES Shader Compiler Version: E031.29.00.00
09-09 18:15:59.783  5269  5296 I Adreno-EGL: Build Date: 04/04/16 Mon
09-09 18:15:59.783  5269  5296 I Adreno-EGL: Local Branch: mybranch19053788
09-09 18:15:59.783  5269  5296 I Adreno-EGL: Remote Branch: quic/LA.BF.1.1.3_rb1.12
09-09 18:15:59.783  5269  5296 I Adreno-EGL: Local Patches: NONE
09-09 18:15:59.783  5269  5296 I Adreno-EGL: Reconstruct Branch: NOTHING
09-09 18:15:59.802  5269  5296 W Adreno-EGL: <qeglDrvAPI_eglGetConfigAttrib:607>: EGL_BAD_ATTRIBUTE
09-09 18:15:59.817  5269  5296 W Adreno-EGL: <qeglDrvAPI_eglGetConfigAttrib:607>: EGL_BAD_ATTRIBUTE
09-09 18:15:59.860  5269  5296 W Gralloc3: mapper 3.x is not supported
09-09 18:15:59.912  5269  5314 W Adreno-EGL: <qeglDrvAPI_eglGetConfigAttrib:607>: EGL_BAD_ATTRIBUTE
09-09 18:15:59.933  5269  5314 I mbgl    : {GLThread 587}[General]: GPU Identifier: Adreno (TM) 330
09-09 18:15:59.933  5269  5314 W mbgl    : {GLThread 587}[OpenGL]: Not using Vertex Array Objects
09-09 18:15:59.953  5269  5269 W robox.liberari: Core platform API violation: Ljava/nio/Buffer;->address:J from Landroid/graphics/Bitmap; using JNI
09-09 18:15:59.953  5269  5269 W robox.liberari: Core platform API violation: Ljava/nio/NIOAccess;->getBaseArrayOffset(Ljava/nio/Buffer;)I from Landroid/graphics/Bitmap; using JNI
09-09 18:15:59.953  5269  5269 W robox.liberari: Core platform API violation: Ljava/nio/NIOAccess;->getBaseArray(Ljava/nio/Buffer;)Ljava/lang/Object; from Landroid/graphics/Bitmap; using JNI
09-09 18:16:04.158  5269  5296 W Adreno-EGL: <qeglDrvAPI_eglGetConfigAttrib:607>: EGL_BAD_ATTRIBUTE
09-09 18:16:04.391  5269  5296 W Adreno-EGL: <qeglDrvAPI_eglGetConfigAttrib:607>: EGL_BAD_ATTRIBUTE
09-09 18:16:06.547  5269  5296 D OpenGLRenderer: endAllActiveAnimators on 0x82c41200 (DropDownListView) with handle 0x8155f700
09-09 18:16:06.658  5269  5269 W IInputConnectionWrapper: getSelectedText on inactive InputConnection
09-09 18:16:06.760  5269  5269 W IInputConnectionWrapper: requestCursorAnchorInfo on inactive InputConnection
09-09 18:16:06.775  5269  5269 W IInputConnectionWrapper: getTextBeforeCursor on inactive InputConnection
09-09 18:16:06.780  5269  5269 W IInputConnectionWrapper: getTextBeforeCursor on inactive InputConnection
09-09 18:16:06.923  5269  5443 W System.err: SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
09-09 18:16:06.946  5269  5443 W System.err: SLF4J: Defaulting to no-operation (NOP) logger implementation
09-09 18:16:06.947  5269  5443 W System.err: SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
09-09 18:16:06.975  5269  5443 D de.grobox.transportr.departures.DeparturesLoader: Departures (12): 3000010
09-09 18:16:06.981  5269  5443 D de.grobox.transportr.departures.DeparturesLoader: Date: Wed Sep 09 18:16:06 GMT+02:00 2020
09-09 18:16:07.386  5269  5269 W ActivityThread: handleWindowVisibility: no activity for token android.os.BinderProxy@4bcb161
09-09 18:16:07.490  5269  5296 W Adreno-EGL: <qeglDrvAPI_eglGetConfigAttrib:607>: EGL_BAD_ATTRIBUTE
09-09 18:16:08.269  5269  5296 W Adreno-EGL: <qeglDrvAPI_eglGetConfigAttrib:607>: EGL_BAD_ATTRIBUTE
09-09 18:16:08.478  5269  5296 W Adreno-EGL: <qeglDrvAPI_eglGetConfigAttrib:607>: EGL_BAD_ATTRIBUTE
09-09 18:16:09.425  5269  5296 D OpenGLRenderer: endAllActiveAnimators on 0x83137600 (DropDownListView) with handle 0x825eaa90
09-09 18:16:09.472  5269  5269 I TripsRepository: From: Location{STATION, 3006904, 50.0014360/8.2584440, place=Mainz, name=Hauptbahnhof, products=[HIGH_SPEED_TRAIN, REGIONAL_TRAIN, SUBURBAN_TRAIN, TRAM, BUS]}
09-09 18:16:09.472  5269  5269 I TripsRepository: Via: null
09-09 18:16:09.472  5269  5269 I TripsRepository: To: Location{STATION, 3000010, 50.1068080/8.6626530, place=Frankfurt (Main), name=Hauptbahnhof, products=[HIGH_SPEED_TRAIN, REGIONAL_TRAIN, SUBURBAN_TRAIN, SUBWAY, TRAM, BUS]}
09-09 18:16:09.473  5269  5269 I TripsRepository: Date: Wed Sep 09 18:16:09 GMT+02:00 2020
09-09 18:16:09.473  5269  5269 I TripsRepository: Departure: true
09-09 18:16:09.473  5269  5269 I TripsRepository: Products: [HIGH_SPEED_TRAIN, REGIONAL_TRAIN, SUBURBAN_TRAIN, SUBWAY, TRAM, BUS, FERRY, CABLECAR, ON_DEMAND]
09-09 18:16:09.473  5269  5269 I TripsRepository: Optimize for: LEAST_DURATION
09-09 18:16:09.473  5269  5269 I TripsRepository: Walk Speed: FAST
09-09 18:16:09.495  5269  5269 W IInputConnectionWrapper: getSelectedText on inactive InputConnection
09-09 18:16:09.527  5269  5269 W IInputConnectionWrapper: requestCursorAnchorInfo on inactive InputConnection
09-09 18:16:09.539  5269  5269 W IInputConnectionWrapper: getTextBeforeCursor on inactive InputConnection
09-09 18:16:09.544  5269  5269 W IInputConnectionWrapper: getTextBeforeCursor on inactive InputConnection
09-09 18:16:09.556  5269  5269 W IInputConnectionWrapper: getTextBeforeCursor on inactive InputConnection
09-09 18:16:09.557  5269  5269 W IInputConnectionWrapper: getSelectedText on inactive InputConnection
09-09 18:16:09.558  5269  5269 W IInputConnectionWrapper: requestCursorAnchorInfo on inactive InputConnection
09-09 18:16:09.558  5269  5269 W IInputConnectionWrapper: getTextBeforeCursor on inactive InputConnection
09-09 18:16:09.562  5269  5269 W IInputConnectionWrapper: getTextBeforeCursor on inactive InputConnection
09-09 18:16:09.687  5269  5281 I robox.liberari: Background young concurrent copying GC freed 46922(2579KB) AllocSpace objects, 75(1664KB) LOS objects, 49% free, 4680KB/9221KB, paused 180us total 176.668ms
09-09 18:16:11.493  5269  5269 W ActivityThread: handleWindowVisibility: no activity for token android.os.BinderProxy@f0009
09-09 18:16:11.536  5269  5269 E ConstraintLayout: layout_constraintWidth_default="wrap" is deprecated.
09-09 18:16:11.536  5269  5269 E ConstraintLayout: Use layout_width="WRAP_CONTENT" and layout_constrainedWidth="true" instead.
09-09 18:16:11.579  5269  5296 W Adreno-EGL: <qeglDrvAPI_eglGetConfigAttrib:607>: EGL_BAD_ATTRIBUTE
09-09 18:16:11.700  5269  5542 W Adreno-EGL: <qeglDrvAPI_eglGetConfigAttrib:607>: EGL_BAD_ATTRIBUTE
09-09 18:16:11.712  5269  5542 W mbgl    : {GLThread 693}[OpenGL]: Not using Vertex Array Objects
09-09 18:16:12.044  5269  5281 I robox.liberari: Background concurrent copying GC freed 73576(3896KB) AllocSpace objects, 29(1872KB) LOS objects, 57% free, 4550KB/10MB, paused 162us total 185.415ms
09-09 18:16:16.305  5269  5281 I robox.liberari: NativeAlloc concurrent copying GC freed 31412(2471KB) AllocSpace objects, 18(988KB) LOS objects, 58% free, 4310KB/10MB, paused 111us total 102.789ms
09-09 18:16:17.972  5269  5296 W Adreno-EGL: <qeglDrvAPI_eglGetConfigAttrib:607>: EGL_BAD_ATTRIBUTE
09-09 18:16:18.693  5269  5269 W ActivityThread: handleWindowVisibility: no activity for token android.os.BinderProxy@e9fb676
09-09 18:16:18.724  5269  5269 E ConstraintLayout: layout_constraintWidth_default="wrap" is deprecated.
09-09 18:16:18.724  5269  5269 E ConstraintLayout: Use layout_width="WRAP_CONTENT" and layout_constrainedWidth="true" instead.
09-09 18:16:18.763  5269  5296 W Adreno-EGL: <qeglDrvAPI_eglGetConfigAttrib:607>: EGL_BAD_ATTRIBUTE
09-09 18:16:18.870  5269  5634 W Adreno-EGL: <qeglDrvAPI_eglGetConfigAttrib:607>: EGL_BAD_ATTRIBUTE
09-09 18:16:18.883  5269  5634 W mbgl    : {GLThread 809}[OpenGL]: Not using Vertex Array Objects
09-09 18:16:19.197  5269  5281 I robox.liberari: Background young concurrent copying GC freed 32764(2445KB) AllocSpace objects, 32(2128KB) LOS objects, 42% free, 5971KB/10MB, paused 168us total 132.094ms
09-09 18:16:20.276  5269  5269 D AndroidRuntime: Shutting down VM
09-09 18:16:20.277  5269  5269 E AndroidRuntime: FATAL EXCEPTION: main
09-09 18:16:20.277  5269  5269 E AndroidRuntime: Process: de.grobox.liberario, PID: 5269
09-09 18:16:20.277  5269  5269 E AndroidRuntime: java.lang.IllegalStateException
09-09 18:16:20.277  5269  5269 E AndroidRuntime: 	at de.grobox.transportr.trips.detail.TripDetailViewModel.reloadTrip(TripDetailViewModel.kt:108)
09-09 18:16:20.277  5269  5269 E AndroidRuntime: 	at de.grobox.transportr.trips.detail.TripDetailFragment.onMenuItemClick(TripDetailFragment.kt:96)
09-09 18:16:20.277  5269  5269 E AndroidRuntime: 	at androidx.appcompat.widget.Toolbar$1.onMenuItemClick(Toolbar.java:207)
09-09 18:16:20.277  5269  5269 E AndroidRuntime: 	at androidx.appcompat.widget.ActionMenuView$MenuBuilderCallback.onMenuItemSelected(ActionMenuView.java:779)
09-09 18:16:20.277  5269  5269 E AndroidRuntime: 	at androidx.appcompat.view.menu.MenuBuilder.dispatchMenuItemSelected(MenuBuilder.java:834)
09-09 18:16:20.277  5269  5269 E AndroidRuntime: 	at androidx.appcompat.view.menu.MenuItemImpl.invoke(MenuItemImpl.java:158)
09-09 18:16:20.277  5269  5269 E AndroidRuntime: 	at androidx.appcompat.view.menu.MenuBuilder.performItemAction(MenuBuilder.java:985)
09-09 18:16:20.277  5269  5269 E AndroidRuntime: 	at androidx.appcompat.view.menu.MenuBuilder.performItemAction(MenuBuilder.java:975)
09-09 18:16:20.277  5269  5269 E AndroidRuntime: 	at androidx.appcompat.widget.ActionMenuView.invokeItem(ActionMenuView.java:623)
09-09 18:16:20.277  5269  5269 E AndroidRuntime: 	at androidx.appcompat.view.menu.ActionMenuItemView.onClick(ActionMenuItemView.java:151)
09-09 18:16:20.277  5269  5269 E AndroidRuntime: 	at android.view.View.performClick(View.java:7259)
09-09 18:16:20.277  5269  5269 E AndroidRuntime: 	at android.view.View.performClickInternal(View.java:7236)
09-09 18:16:20.277  5269  5269 E AndroidRuntime: 	at android.view.View.access$3600(View.java:801)
09-09 18:16:20.277  5269  5269 E AndroidRuntime: 	at android.view.View$PerformClick.run(View.java:27892)
09-09 18:16:20.277  5269  5269 E AndroidRuntime: 	at android.os.Handler.handleCallback(Handler.java:883)
09-09 18:16:20.277  5269  5269 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:100)
09-09 18:16:20.277  5269  5269 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:214)
09-09 18:16:20.277  5269  5269 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:7356)
09-09 18:16:20.277  5269  5269 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
09-09 18:16:20.277  5269  5269 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:491)
09-09 18:16:20.277  5269  5269 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:940)
09-09 18:16:26.097  1461  1531 I ActivityManager: Killing 5269:de.grobox.liberario/u0a133 (adj 0): crash
09-09 18:16:26.249  1461  1710 W ActivityManager: setHasOverlayUi called on unknown pid: 5269
09-09 18:16:26.283  1461  1722 I libprocessgroup: Successfully killed process cgroup uid 10133 pid 5269 in 184ms
09-09 18:16:26.291  1435  1435 I Zygote  : Process 5269 exited due to signal 9 (Killed)
```

**Stacktrace number 2**

```bash
$ adb logcat | grep -F "`adb shell ps | grep de.grobox.  | tr -s [:space:] ' ' | cut -d' ' -f2`"
09-09 18:16:00.103  1435  1435 D Zygote  : Forked child process 5325
09-09 18:16:00.121  1461  1721 I ActivityManager: Start proc 5325:de.grobox.liberario:lost/u0a133 for service {de.grobox.liberario/com.mapzen.android.lost.internal.FusedLocationProviderService}
09-09 18:16:00.212  5325  5325 V AndroidLocationEngine: Initializing.
09-09 18:16:00.219  5325  5325 V MapboxTelemetry: Initializing telemetry.
09-09 18:16:00.227  5325  5325 D NetworkSecurityConfig: Using Network Security Config from resource network_security_config debugBuild: false
09-09 18:16:00.276  5325  5325 V MapboxTelemetry: Disabling telemetry.
09-09 18:16:00.277  5325  5325 D MapboxTelemetry: Shutting down telemetry service.
09-09 18:16:00.291  5325  5325 I .liberario:los: Thread[1,tid=5325,Native,Thread*=0xb3100e00,peer=0x723211f0,"main"] recursive attempt to load library "/data/app/de.grobox.liberario-KBS5ZhV_anmfK-co2N5C-Q==/lib/arm/libmapbox-gl.so"
09-09 18:16:00.295  5325  5325 I chatty  : uid=10133 de.grobox.liberario:lost identical 1 line
09-09 18:16:00.296  5325  5325 I .liberario:los: Thread[1,tid=5325,Native,Thread*=0xb3100e00,peer=0x723211f0,"main"] recursive attempt to load library "/data/app/de.grobox.liberario-KBS5ZhV_anmfK-co2N5C-Q==/lib/arm/libmapbox-gl.so"
09-09 18:16:00.303  5325  5325 V MapboxTelemetry: Telemetry was already initialized on that state (enabled: false).
09-09 18:16:00.366  5325  5342 E FusionEngine: Unable to register for network updates.
09-09 18:16:00.366  5325  5342 E FusionEngine: java.lang.IllegalArgumentException: provider doesn't exist: network
09-09 18:16:00.366  5325  5342 E FusionEngine: 	at android.os.Parcel.createException(Parcel.java:2075)
09-09 18:16:00.366  5325  5342 E FusionEngine: 	at android.os.Parcel.readException(Parcel.java:2039)
09-09 18:16:00.366  5325  5342 E FusionEngine: 	at android.os.Parcel.readException(Parcel.java:1987)
09-09 18:16:00.366  5325  5342 E FusionEngine: 	at android.location.ILocationManager$Stub$Proxy.requestLocationUpdates(ILocationManager.java:1151)
09-09 18:16:00.366  5325  5342 E FusionEngine: 	at android.location.LocationManager.requestLocationUpdates(LocationManager.java:1019)
09-09 18:16:00.366  5325  5342 E FusionEngine: 	at android.location.LocationManager.requestLocationUpdates(LocationManager.java:590)
09-09 18:16:00.366  5325  5342 E FusionEngine: 	at com.mapzen.android.lost.internal.FusionEngine.enableNetwork(FusionEngine.java:154)
09-09 18:16:00.366  5325  5342 E FusionEngine: 	at com.mapzen.android.lost.internal.FusionEngine.enable(FusionEngine.java:112)
09-09 18:16:00.366  5325  5342 E FusionEngine: 	at com.mapzen.android.lost.internal.LocationEngine.addRequest(LocationEngine.java:47)
09-09 18:16:00.366  5325  5342 E FusionEngine: 	at com.mapzen.android.lost.internal.FusedLocationProviderServiceDelegate.requestLocationUpdates(FusedLocationProviderServiceDelegate.java:65)
09-09 18:16:00.366  5325  5342 E FusionEngine: 	at com.mapzen.android.lost.internal.FusedLocationProviderService$1.requestLocationUpdates(FusedLocationProviderService.java:43)
09-09 18:16:00.366  5325  5342 E FusionEngine: 	at com.mapzen.android.lost.internal.IFusedLocationProviderService$Stub.onTransact(IFusedLocationProviderService.java:101)
09-09 18:16:00.366  5325  5342 E FusionEngine: 	at android.os.Binder.execTransactInternal(Binder.java:1021)
09-09 18:16:00.366  5325  5342 E FusionEngine: 	at android.os.Binder.execTransact(Binder.java:994)
09-09 18:16:00.366  5325  5342 E FusionEngine: Caused by: android.os.RemoteException: Remote stack trace:
09-09 18:16:00.366  5325  5342 E FusionEngine: 	at com.android.server.LocationManagerService.requestLocationUpdatesLocked(LocationManagerService.java:2575)
09-09 18:16:00.366  5325  5342 E FusionEngine: 	at com.android.server.LocationManagerService.requestLocationUpdates(LocationManagerService.java:2555)
09-09 18:16:00.366  5325  5342 E FusionEngine: 	at android.location.ILocationManager$Stub.onTransact(ILocationManager.java:468)
09-09 18:16:00.366  5325  5342 E FusionEngine: 	at android.os.Binder.execTransactInternal(Binder.java:1021)
09-09 18:16:00.366  5325  5342 E FusionEngine: 	at android.os.Binder.execTransact(Binder.java:994)
09-09 18:16:00.366  5325  5342 E FusionEngine: 
09-09 18:16:00.373  5325  5343 E FusionEngine: Unable to register for network updates.
09-09 18:16:00.373  5325  5343 E FusionEngine: java.lang.IllegalArgumentException: provider doesn't exist: network
09-09 18:16:00.373  5325  5343 E FusionEngine: 	at android.os.Parcel.createException(Parcel.java:2075)
09-09 18:16:00.373  5325  5343 E FusionEngine: 	at android.os.Parcel.readException(Parcel.java:2039)
09-09 18:16:00.373  5325  5343 E FusionEngine: 	at android.os.Parcel.readException(Parcel.java:1987)
09-09 18:16:00.373  5325  5343 E FusionEngine: 	at android.location.ILocationManager$Stub$Proxy.requestLocationUpdates(ILocationManager.java:1151)
09-09 18:16:00.373  5325  5343 E FusionEngine: 	at android.location.LocationManager.requestLocationUpdates(LocationManager.java:1019)
09-09 18:16:00.373  5325  5343 E FusionEngine: 	at android.location.LocationManager.requestLocationUpdates(LocationManager.java:590)
09-09 18:16:00.373  5325  5343 E FusionEngine: 	at com.mapzen.android.lost.internal.FusionEngine.enableNetwork(FusionEngine.java:154)
09-09 18:16:00.373  5325  5343 E FusionEngine: 	at com.mapzen.android.lost.internal.FusionEngine.enable(FusionEngine.java:112)
09-09 18:16:00.373  5325  5343 E FusionEngine: 	at com.mapzen.android.lost.internal.LocationEngine.addRequest(LocationEngine.java:47)
09-09 18:16:00.373  5325  5343 E FusionEngine: 	at com.mapzen.android.lost.internal.FusedLocationProviderServiceDelegate.requestLocationUpdates(FusedLocationProviderServiceDelegate.java:65)
09-09 18:16:00.373  5325  5343 E FusionEngine: 	at com.mapzen.android.lost.internal.FusedLocationProviderService$1.requestLocationUpdates(FusedLocationProviderService.java:43)
09-09 18:16:00.373  5325  5343 E FusionEngine: 	at com.mapzen.android.lost.internal.IFusedLocationProviderService$Stub.onTransact(IFusedLocationProviderService.java:101)
09-09 18:16:00.373  5325  5343 E FusionEngine: 	at android.os.Binder.execTransactInternal(Binder.java:1021)
09-09 18:16:00.373  5325  5343 E FusionEngine: 	at android.os.Binder.execTransact(Binder.java:994)
09-09 18:17:52.627  5325  5349 E FusionEngine: Unable to register for network updates.
09-09 18:17:52.627  5325  5349 E FusionEngine: java.lang.IllegalArgumentException: provider doesn't exist: network
09-09 18:17:52.627  5325  5349 E FusionEngine: 	at android.os.Parcel.createException(Parcel.java:2075)
09-09 18:17:52.627  5325  5349 E FusionEngine: 	at android.os.Parcel.readException(Parcel.java:2039)
09-09 18:17:52.627  5325  5349 E FusionEngine: 	at android.os.Parcel.readException(Parcel.java:1987)
09-09 18:17:52.627  5325  5349 E FusionEngine: 	at android.location.ILocationManager$Stub$Proxy.requestLocationUpdates(ILocationManager.java:1151)
09-09 18:17:52.627  5325  5349 E FusionEngine: 	at android.location.LocationManager.requestLocationUpdates(LocationManager.java:1019)
09-09 18:17:52.627  5325  5349 E FusionEngine: 	at android.location.LocationManager.requestLocationUpdates(LocationManager.java:590)
09-09 18:17:52.627  5325  5349 E FusionEngine: 	at com.mapzen.android.lost.internal.FusionEngine.enableNetwork(FusionEngine.java:154)
09-09 18:17:52.627  5325  5349 E FusionEngine: 	at com.mapzen.android.lost.internal.FusionEngine.enable(FusionEngine.java:112)
09-09 18:17:52.627  5325  5349 E FusionEngine: 	at com.mapzen.android.lost.internal.LocationEngine.addRequest(LocationEngine.java:47)
09-09 18:17:52.627  5325  5349 E FusionEngine: 	at com.mapzen.android.lost.internal.FusedLocationProviderServiceDelegate.requestLocationUpdates(FusedLocationProviderServiceDelegate.java:65)
09-09 18:17:52.627  5325  5349 E FusionEngine: 	at com.mapzen.android.lost.internal.FusedLocationProviderService$1.requestLocationUpdates(FusedLocationProviderService.java:43)
09-09 18:17:52.627  5325  5349 E FusionEngine: 	at com.mapzen.android.lost.internal.IFusedLocationProviderService$Stub.onTransact(IFusedLocationProviderService.java:101)
09-09 18:17:52.627  5325  5349 E FusionEngine: 	at android.os.Binder.execTransactInternal(Binder.java:1021)
09-09 18:17:52.627  5325  5349 E FusionEngine: 	at android.os.Binder.execTransact(Binder.java:994)
09-09 18:17:52.627  5325  5349 E FusionEngine: Caused by: android.os.RemoteException: Remote stack trace:
09-09 18:17:52.627  5325  5349 E FusionEngine: 	at com.android.server.LocationManagerService.requestLocationUpdatesLocked(LocationManagerService.java:2575)
09-09 18:17:52.627  5325  5349 E FusionEngine: 	at com.android.server.LocationManagerService.requestLocationUpdates(LocationManagerService.java:2555)
09-09 18:17:52.627  5325  5349 E FusionEngine: 	at android.location.ILocationManager$Stub.onTransact(ILocationManager.java:468)
09-09 18:17:52.627  5325  5349 E FusionEngine: 	at android.os.Binder.execTransactInternal(Binder.java:1021)
09-09 18:17:52.627  5325  5349 E FusionEngine: 	at android.os.Binder.execTransact(Binder.java:994)
09-09 18:17:52.627  5325  5349 E FusionEngine: 
09-09 18:17:52.632  5325  5342 E FusionEngine: Unable to register for network updates.
09-09 18:17:52.632  5325  5342 E FusionEngine: java.lang.IllegalArgumentException: provider doesn't exist: network
09-09 18:17:52.632  5325  5342 E FusionEngine: 	at android.os.Parcel.createException(Parcel.java:2075)
09-09 18:17:52.632  5325  5342 E FusionEngine: 	at android.os.Parcel.readException(Parcel.java:2039)
09-09 18:17:52.632  5325  5342 E FusionEngine: 	at android.os.Parcel.readException(Parcel.java:1987)
09-09 18:17:52.632  5325  5342 E FusionEngine: 	at android.location.ILocationManager$Stub$Proxy.requestLocationUpdates(ILocationManager.java:1151)
09-09 18:17:52.632  5325  5342 E FusionEngine: 	at android.location.LocationManager.requestLocationUpdates(LocationManager.java:1019)
09-09 18:17:52.632  5325  5342 E FusionEngine: 	at android.location.LocationManager.requestLocationUpdates(LocationManager.java:590)
09-09 18:17:52.632  5325  5342 E FusionEngine: 	at com.mapzen.android.lost.internal.FusionEngine.enableNetwork(FusionEngine.java:154)
09-09 18:17:52.632  5325  5342 E FusionEngine: 	at com.mapzen.android.lost.internal.FusionEngine.enable(FusionEngine.java:112)
09-09 18:17:52.632  5325  5342 E FusionEngine: 	at com.mapzen.android.lost.internal.LocationEngine.addRequest(LocationEngine.java:47)
09-09 18:17:52.632  5325  5342 E FusionEngine: 	at com.mapzen.android.lost.internal.FusedLocationProviderServiceDelegate.requestLocationUpdates(FusedLocationProviderServiceDelegate.java:65)
09-09 18:17:52.632  5325  5342 E FusionEngine: 	at com.mapzen.android.lost.internal.FusedLocationProviderService$1.requestLocationUpdates(FusedLocationProviderService.java:43)
09-09 18:17:52.632  5325  5342 E FusionEngine: 	at com.mapzen.android.lost.internal.IFusedLocationProviderService$Stub.onTransact(IFusedLocationProviderService.java:101)
09-09 18:17:52.632  5325  5342 E FusionEngine: 	at android.os.Binder.execTransactInternal(Binder.java:1021)
09-09 18:17:52.632  5325  5342 E FusionEngine: 	at android.os.Binder.execTransact(Binder.java:994)
```

Commands as per [this Stackoverflow answer](https://stackoverflow.com/questions/6854127/filter-logcat-to-get-only-the-messages-from-my-application-in-android/9869609#9869609).

--- a/app/src/main/java/de/grobox/transportr/trips/search/DirectionsFragment.kt
+++ b/app/src/main/java/de/grobox/transportr/trips/search/DirectionsFragment.kt
@@ -246,7 +246,7 @@ class DirectionsFragment : TransportrFragment() {
 
     private fun onFindGpsLocation(type: FavLocationType?) {
         if (type == null) {
-            viewModel.locationLiveData.removeObservers(this@DirectionsFragment)
+            viewModel.locationLiveData.removeObservers(viewLifecycleOwner)
             fromLocation.clearSearching()
             return
         }
@@ -255,10 +255,10 @@ class DirectionsFragment : TransportrFragment() {
         }
         fromLocation.setSearching()
         toLocation.requestFocus()
-        viewModel.locationLiveData.observe(viewLifecycleOwner, Observer { location ->
+        viewModel.locationLiveData.observe(viewLifecycleOwner, { location ->
             viewModel.setFromLocation(location)
             viewModel.search()
-            viewModel.locationLiveData.removeObservers(this@DirectionsFragment)
+            viewModel.locationLiveData.removeObservers(viewLifecycleOwner)
         })
     }
 

